### PR TITLE
Add calendar manual te creation on long press for droid

### LINKED ISSUE #5868: Calendar: Add manual time entry creation
Depends on #5867 

Manual TEs
- [x] Use long press to create new Entry with handlers and contextual menu appear.
- [x] New entries should snap to quarter hour and other calendar events start and end times
- [x] Should have a default duration of 30 min
- [x] If created in the gap between 2 TEs that is 2 hours or less, it should fill the whole gap by default

Design: https://app.zeplin.io/project/5a0d3cd27a75add1a8924f6e/screen/5d8a2802a3c39b61c2643a26

### PR DESCRIPTION
**Pull request by daividssilverio**
## What's this?
This PR enables manual time entry creation when you long press in the calendar view.

### Relationships
Closes #5868

## How is it done?
It uses the `calendarLayoutCalculator.CalculateTwoHoursOrLessGapsLayoutAttributes` to find the gaps, selecting the start and end times for manually created time entries on long presses.

Then the responsibility is forwarded to the contextual menu view model. The calendar view simply reacts to a new calendar item in edit mode and displays it regularly.

## :squid: Permissions
📆 pals

### CODE REVIEW

File: Toggl.Droid/Views/Calendar/CalendarDayView.cs [hunk 1]
```diff
@@ -295,7 +326,25 @@ private void onLongPress(MotionEvent e1)
...
+                foreach (var gap in teGaps)
+                {
+                    if (startTime > gap.StartTime && startTime < gap.StartTime + gap.Duration)
+                    {
+                        duration = gap.Duration;
+                        startTime = gap.StartTime;
+                        break;
+                    }
+                }
+            }
```
Comment (semanticer) [inline Toggl.Droid/Views/Calendar/CalendarDayView.cs]: Maybe less imperative looks better? 
```csharp
var gap = teGaps.Find((g) => startTime > g.StartTime && startTime < g.StartTime + g.Duration);
duration = gap.Duration;
startTime = gap.StartTime;
```
Maybe I don't see some edge case or something.

Comment (daividssilverio) [reply Toggl.Droid/Views/Calendar/CalendarDayView.cs]: The edge case is that we might not find any gaps.  Let me give it a try.

File: Toggl.Droid/Views/Calendar/CalendarDayView.cs [hunk 2]
```diff
@@ -142,21 +142,52 @@ public void SetCurrentDate(DateTimeOffset dateOnView)
 
         public void SetCurrentItemInEditMode(CalendarItem? calendarItemInEditMode)
         {
+            var isCurrentlyEditingItem = isEditingItem();
             if (!calendarItemInEditMode.HasValue)
```
Comment (heytherewill) [inline Toggl.Droid/Views/Calendar/CalendarDayView.cs]: Why not comparing it to null?

File: Toggl.Droid/Views/Calendar/CalendarDayView.cs [hunk 3]
```diff
@@ -142,21 +142,52 @@ public void SetCurrentDate(DateTimeOffset dateOnView)
...
             {
-                if (isEditingItem())
+                if (isCurrentlyEditingItem)
```
Comment (heytherewill) [inline Toggl.Droid/Views/Calendar/CalendarDayView.cs]: Please add braces even for single line statements that are not early returns

File: Toggl.Droid/Views/Calendar/CalendarDayView.cs [hunk 4]
```diff
@@ -295,7 +326,25 @@ private void onLongPress(MotionEvent e1)
             var touchY = e1.GetY();
             var calendarItemInfo = findCalendarItemFromPoint(touchX, touchY);
             
-            //todo: do something onLongPress?
+            if (calendarItemInfo.IsValid) return;
```
Comment (heytherewill) [inline Toggl.Droid/Views/Calendar/CalendarDayView.cs]: Returns in new lines

File: Toggl.Droid/Views/Calendar/CalendarDayView.cs (hunk 2 above)
Comment (daividssilverio) [reply Toggl.Droid/Views/Calendar/CalendarDayView.cs]: this reads better, is there any advantage to comparing with null?

File: Toggl.Droid/Views/Calendar/CalendarDayView.cs (hunk 3 above)
Comment (daividssilverio) [reply Toggl.Droid/Views/Calendar/CalendarDayView.cs]: I've fixed this #6334 for convenience.

File: Toggl.Droid/Views/Calendar/CalendarDayView.cs (hunk 4 above)
Comment (daividssilverio) [reply Toggl.Droid/Views/Calendar/CalendarDayView.cs]: I've fixed this #6334 for convenience.

**Pull request merged**